### PR TITLE
Daniel Cogan

### PR DESCRIPTION
**Pull request by danield9tqh**

### CODE REVIEW

File: context-free.md [hunk 1]
```diff
@@ -1,22 +1,22 @@
 # Context Free
 
 ##  Who is this programming language for?
-
++ This language is for people who want to programatically create images. 
 
 ## What is easy to do in this language? Why is it easy?
-
++ It is easy to draw images that are made up of many much simpler shapes such as squares, circles etc. This is easy because the definitions for these shapes are already available in the language.
```
Comment (jkutsko) [inline context-free.md:7]: I would additionally mention that its easy to make these shapes defined by recurrence relations, so its not just that its easy to make lots of the same triangle.

File: context-free.md [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
 ## How did you learn how to program in this language?
-
++ I looked at the online documentation and also at the example welcome image shown when the program first started. I also experimented with different code to see what certain functions did. 
 
 ## What is the underlying _computational model_ for this programming language? 
 _We don't yet have a great definition of the term "computational model". 
 For now, try to come up with the clearest, most concise explanation of what 
 happens when a ContextFree program runs._
-
++ Most likely, the ContextFree program starts at the startshape and successively renders each subimage in that startshape. It then renders each sub-sub image and continues to do this until it gets to a base image, a square or a circle, which it draws on the canvas.
```
Comment (jkutsko) [inline context-free.md:19]: I'm not sure about this myself, but its an interesting question - In what order does the program choose to render each sub image? If you have 2 fractals in an image, does it render all of one of them and then the other, or does it render both one layer at a time?

File: context-free.md [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
 ## What is the underlying _computational model_ for this programming language? 
 _We don't yet have a great definition of the term "computational model". 
 For now, try to come up with the clearest, most concise explanation of what 
 happens when a ContextFree program runs._
-
++ Most likely, the ContextFree program starts at the startshape and successively renders each subimage in that startshape. It then renders each sub-sub image and continues to do this until it gets to a base image, a square or a circle, which it draws on the canvas. 
 
 ## What do you think is interesting about the ContextFree program you wrote?
-
++ My program is intereting because it takes advantage of the subroutine model of ContextFree by repeating similar shapes but being able to make very slight changes such as color or orientation.
```
Comment (jkutsko) [inline context-free.md:22]: How did you come up with your design? I like how you made it simple in principle, but look interesting. Did you play around with anything similar?

File: my-dsl.md [hunk 4]
```diff
@@ -2,37 +2,41 @@
...
 # Domain
 _Describe the language's domain in five words._
 
-
-# Computational model
-_We don't yet have a great definition of the term "computational model". 
-For now, try to come up with the clearest, most concise explanation of 
-what happens when a program in your DSL runs._
-
++ Searching / Querying application log data
```
Comment (jkutsko) [inline my-dsl.md:13]: I think you accidentally deleted the question about the computational model of your language.

File: my-dsl.md [hunk 5]
```diff
@@ -2,37 +2,41 @@
...
 # Host language
 _What language(s) was (were) used to implement the DSL?_
 
++ The language is based mostly on SQL queries and syntax and also include Regex for matching log messages. 
 
 # Benefits
 _Identify one potential benefit of the DSL: how is a programmer's life or a 
 company's bottom line made easier by the existence of this language?_
 
++ One benefit of this language is that a company can search its application log messages through a language with a low learning curve. Because the language is based off of SQL syntax, it is easy to understand but at the same time provides funtionality specific for searching log messages.
```
Comment (jkutsko) [inline my-dsl.md:36]: Sounds cool, and definitely seems like a useful domain to have a language for.

File: context-free.md [hunk 6]
```diff
@@ -1,22 +1,22 @@
 # Context Free
 
 ##  Who is this programming language for?
-
++ This language is for people who want to programatically create images.
```
Comment (aputman) [inline context-free.md:4]: Does the different features of the language change who might want to use it most?

File: context-free.md (hunk 1 above)
Comment (aputman) [inline context-free.md:7]: +1

File: my-dsl.md [hunk 7]
```diff
@@ -2,37 +2,41 @@
...
-what happens when a program in your DSL runs._
-
++ Searching / Querying application log data
 
 # DSL-ness
 _Fowler writes about a spectrum of languages, from general-purpose languages to 
 "purely" domain-specific. Where does the DSL you chose fall on this spectrum, 
 and why?_ 
 
++ I would classify this language as purely domain-specific because it is not turing complete. It also lacks many features that Fowler points out are necessary for a general-purpose language such as, "conditions, looping constructs and variables."
```
Comment (aputman) [inline my-dsl.md:19]: Are there any features it has that might make it close to the general-purpose side?

**Pull request closed without merging**